pkg/maps/ctmap/ctmap.go

```diff
@@ -627,6 +620,21 @@ func PurgeOrphanNATEntries(ctMapTCP, ctMapAny *Map) *NatGCStats {
 				}
 			} else {
 				stats.IngressAlive += 1
+				stats.EgressAlive += 1
+			}
+		} else if natKey.GetFlags()&tuple.TUPLE_F_OUT == tuple.TUPLE_F_OUT {
+			ingressNatKey := ingressNatKeyFromEgressNatKeyVal(natKey, natVal)
```

Comment:
For my understanding - instead of checking for "no Ingress CT entry AND no Ingress NAT entry", we could also check for "no Ingress CT entry AND no Egress CT entry". Correct?

Should we raise some warning if we find an Ingress CT entry, but the entry doesn't have the DSR flag?

Reply:
>For my understanding - instead of checking for "no Ingress CT entry AND no Ingress NAT entry", we could also check for "no Ingress CT entry AND no Egress CT entry". Correct?

Yep, that is correct also to my understanding :+1: 

I decided to go for the CT and NAT entries check to make it more explicit that (i.e. it's a DSR SNAT entry and it doesn't have a related NAT entry), but perhaps once we are happy with this we can evaluate the incress/egress CT check and maybe unify the 2 cases (just food for thoughts)

> Should we raise some warning if we find an Ingress CT entry, but the entry doesn't have the DSR flag?

I'm not sure here, I would imagine it's possible to have ingress CT entries without NAT being involved

Reply:
> > For my understanding - instead of checking for "no Ingress CT entry AND no Ingress NAT entry", we could also check for "no Ingress CT entry AND no Egress CT entry". Correct?
> 
> Yep, that is correct also to my understanding +1
> 
> I decided to go for the CT and NAT entries check to make it more explicit (i.e. it's a DSR SNAT entry and it doesn't have a related NAT entry), 

yeah, that's what I figured :). I understand where you're coming from (focusing it exactly on how the situation looks like for DSR). But I feel the reasoning "if there's no ingress CT and no egress CT, then obviously this NAT entry is stale" is more straight-forward, and requires less understanding of how NAT/CT works for DSR in detail. Maybe that tips the balance, both approaches being equal? Either way is alright.


> > Should we raise some warning if we find an Ingress CT entry, but the entry doesn't have the DSR flag?
> 
> I'm not sure here, I would imagine it's possible to have ingress CT entries without NAT being involved

Ah sorry. I specifically meant for the "we are dealing with a DSR entry" path. Here it would be nice to know if we unexpectedly match something that's **not** DSR.